x/mint/keeper/last_block_info.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/archway-network/archway/x/mint/types"
+)
+
+// SetLastBlockInfo stores the last block info. Fails and returns an error if the input is not valid
+func (k Keeper) SetLastBlockInfo(ctx sdk.Context, lbi types.LastBlockInfo) error {
+	if err := lbi.Validate(); err != nil {
+		return err
+	}
+	store := ctx.KVStore(k.storeKey)
+	value := k.cdc.MustMarshal(&lbi)
+
+	store.Set(types.LastBlockInfoPrefix, value)
+	return nil
+}
+
+// GetLastBlockInfo fetches the latest block info
+func (k Keeper) GetLastBlockInfo(ctx sdk.Context) (bool, types.LastBlockInfo) {
```

Comment:
here we should invert the return types to: `types.LastBlockInfo, bool`, it's more go-idiomatic.